Sound/Tidal/EspGrid.hs

```diff
@@ -0,0 +1,114 @@
+module Sound.Tidal.EspGrid where
+
+import Control.Concurrent.MVar
+import Control.Concurrent
+import Control.Monad (forever)
+import Control.Monad.Loops (iterateM_)
+import Sound.OSC.FD
+import Data.Time.Clock
+import Data.Time.Clock.POSIX
+import Data.Time.Calendar (fromGregorian)
+
+import Sound.Tidal.Tempo
+import Sound.Tidal.Time as T
+import Sound.Tidal.Stream
+import Sound.Tidal.Dirt
+import Sound.Tidal.Transition (transition)
+import Sound.Tidal.Pattern (silence)
+
+parseEspTempo :: [Datum] -> Maybe Tempo
+parseEspTempo d = do

```

Comment:
One difference though would be that with the "one by one" deindexing it would work were the message/list to gain extra data, whereas with the pattern matching it is tolerant to the addition of extra information to the message. I don't have strong opinions about it, but it is not just a question of idiom...